src/components/refactor/dialogs/shunt-compensator-creation/shunt-compensator-creation-dialog.js

```diff
@@ -0,0 +1,277 @@
+        [MAXIMUM_NUMBER_OF_SECTIONS]: yup
+            .number()
+            .moreThan(0, 'ShuntCompensatorErrorMaximumLessThanOne')
+            .required(),
+        [CURRENT_NUMBER_OF_SECTIONS]: yup
+            .number()
+            .min(0, 'ShuntCompensatorErrorCurrentLessThanZero')
+            .max(
+                yup.ref(MAXIMUM_NUMBER_OF_SECTIONS),
+                'ShuntCompensatorErrorCurrentLessThanMaximum'
+            )
+            .required(),
+        [IDENTICAL_SECTIONS]: yup.bool().required(),
+        [SUSCEPTANCE_PER_SECTION]: yup.number().nullable().required(),
+        ...getConnectivityFormValidationSchema(),
+    })
+    .required();
+
+/**
+ * Dialog to create a shunt compensator in the network
+ * @param voltageLevelOptionsPromise Promise handling list of voltage level options
+ * @param currentNodeUuid the node we are currently working on
+ * @param editData the data to edit
+ * @param dialogProps props that are forwarded to the generic ModificationDialog component
+ */
+const ShuntCompensatorCreationDialog = ({
+    voltageLevelOptionsPromise,
+    currentNodeUuid,
+    editData,
+    ...dialogProps
+}) => {
+    const studyUuid = decodeURIComponent(useParams().studyUuid);
+
+    const { snackError } = useSnackMessage();
+
+    const equipmentPath = 'shunt-compensators';
+
+    const methods = useForm({
+        defaultValues: emptyFormData,
+        resolver: yupResolver(schema),
+    });
+
+    const { reset } = methods;
+
+    const fromSearchCopyToFormValues = (shuntCompensator) => {
+        fetchEquipmentInfos(
+            studyUuid,
+            currentNodeUuid,
+            'voltage-levels',
+            shuntCompensator.voltageLevelId,
+            true
+        ).then((vlResult) => {
+            reset({
+                [EQUIPMENT_ID]: shuntCompensator.id + '(1)',
+                [EQUIPMENT_NAME]: shuntCompensator.name ?? '',
+                [MAXIMUM_NUMBER_OF_SECTIONS]:
+                    shuntCompensator.maximumSectionCount,
+                [CURRENT_NUMBER_OF_SECTIONS]: shuntCompensator.sectionCount,
+                [SUSCEPTANCE_PER_SECTION]: shuntCompensator.bperSection,
+                ...getConnectivityFormData({
+                    voltageLevelId: shuntCompensator.voltageLevelId,
+                    voltageLevelTopologyKind: vlResult.topologyKind,
+                    voltageLevelName: vlResult.name,
+                    voltageLevelNominalVoltage: vlResult.nominalVoltage,
+                    voltageLevelSubstationId: vlResult.substationId,
+                    connectionDirection: shuntCompensator.connectionDirection,
+                    connectionName: shuntCompensator.connectionName,
+                    connectionPosition: shuntCompensator.connectionPosition,
+                }),
+            });
+        });
+    };
+
+    const fromEditDataToFormValues = useCallback(
+        (shuntCompensator) => {
+            fetchEquipmentInfos(
+                studyUuid,
```

Comment:
no need to this method here, you should remove it because the object shuntCompensator contain the wanted information

Reply:
removed